tests/suites/cloud/tests/supervisor/index.js

```diff
@@ -0,0 +1,238 @@
+`use strict`;
+const { exec } = require("mz/child_process");
+
+module.exports = {
+  title: "Supervisor test suite",
+  tests: [
+    {
+      title: "Provisioning without deltas",
+      run: async function (test) {
+        // push multicontainer app release to new app
+        test.comment(`Cloning repo...`);
+        await exec(
+          `git clone https://github.com/balena-io-examples/balena-python-hello-world.git ${__dirname}/app`
+        );
+
+        test.comment(`Pushing release...`);
+
+        let firstCommit = await this.context.get().cloud.pushReleaseToApp(
+          this.context.get().balena.application, 
+          `${__dirname}/app`
+        );
+        test.comment(`Application commit is ${firstCommit}`);
+
+        // wait until we have the application downloaded
+        await this.context.get().cloud.waitUntilServicesRunning(
+          this.context.get().balena.uuid, 
+          [`main`], 
+          firstCommit
+        )
+
+        test.comment(`Disabling deltas`);
+        await this.context
+          .get()
+          .cloud.balena.models.device.configVar.set(
+            this.context.get().balena.uuid,
+            "RESIN_SUPERVISOR_DELTA",
+            0
+          );
+
+        // add a comment to the end of the main.py file, to trigger a delta
+        await exec(`echo "#comment" >> ${__dirname}/app/src/main.py`);
+        test.comment(`Pushing release...`);
+
+        let secondCommit = await this.context.get().cloud.pushReleaseToApp(
+          this.context.get().balena.application, 
+          `${__dirname}/app`
+        );
+        test.comment(`Application commit is ${secondCommit}`);
+
+        //check device is now on new commit
+        await this.context.get().cloud.waitUntilServicesRunning(
+          this.context.get().balena.uuid, 
+          [`main`], 
+          secondCommit
+        )
+
+        // device should have downloaded application without mentioning that deltas are being used
+        let usedDeltas = await this.context.get().cloud.checkLogsContain(
+          this.context.get().balena.uuid, 
+          `Downloading delta for image`, 
+          `Applied configuration change {"SUPERVISOR_DELTA":"0"}`
+        );
+
+        test.is(
+          !usedDeltas,
+          true,
+          `Device shouldn't use deltas to download new release`
+        );
+      },
+    },
+    {
+      title: "Supervisor reload test",
+      run: async function (test) {
+        let supervisorVersion = await this.context.get().cloud.getSupervisorVersion(this.context.get().balena.uuid)
+        test.comment(`Supervisor version ${supervisorVersion} detected`);
+
+        // remove supervisor container
+        test.comment(`removing supervisor`);
+        await this.context
+          .get()
+          .cloud.executeCommandInHostOS(
+            `systemctl stop balena-supervisor && balena rm balena_supervisor && balena rmi -f $(balena images | grep supervisor | awk '{print $3}')`,
```

Comment:
Removing the old image `balena rmi ...` is required for now due to some bad logic in the update supervisor script and should be fixed by https://github.com/balena-os/meta-balena/pull/2180.

Once that is merged, we will actually want to remove this `balena rmi` completely. Once supervisor starts getting treatment like a standard app the image names will not have unique names like `supervisor`.

Maybe worth a comment, but needs to stay here for now.

Reply:
This is per `TC26 - Supervisor reload test` and is going to change once we merge https://github.com/balena-os/meta-balena/pull/2336